src/throwIf.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * throw return of `err` if predicate function return true
+ *
+ * @example
+ * ```ts
+ *  pipe(
+ *    fn(), // return type is string | undefined
+ *
+ *    throwIf(isUndefined)((input) => Error("return of fn is undefined")),
+ *    // input is undefined, and it is return of fn
+ *
+ *    (input) => input, // input is string
+ *  )
+ * ```
+ *
+ * @param predicate typeguard function for T
+ * @return T exclude N
+ */
```

Comment:
Considering usability, it seems like predicates should return a boolean. This is because JavaScript allows implicit boolean conversions for all values in if statements, but the results are inconsistent. Due to this, and the fact that understanding JavaScript's language-specific features is required to predict the outcome accurately, it would be better to restrict predicates to functions that return boolean (or use type guards).

Reply:
Also, considering that the argument 'err' accepts "input" as its parameter, would it be beneficial to use the `identity` function as the default value?

Using the identity function as the default value could be a good option.

Reply:
Alright. I agree with your opinion, and anyway, the main point of my suggestion is to allow variadic arguments. It seems better to continue working on this in a separate PR, apart from this one. In this PR, please only keep the code related to the support of generic types in custom-typeguard functions, and please remove the code for adding new functions (`isDefined` and `throwIf`) for now.